components/brave_wallet/common/eth_abi_utils.cc

```diff
@@ -203,6 +208,67 @@ absl::optional<std::vector<std::string>> ExtractStringArray(Span string_array) {
   return result;
 }
 
+absl::optional<std::pair<bool, std::vector<uint8_t>>> ExtractBoolAndBytes(
+    Span data) {
+  auto bool_row = ExtractRow(data, 0);
+  if (!bool_row) {
+    return absl::nullopt;
+  }
+
+  auto bytes = ExtractBytesFromTuple(data, 1);
+  if (!bytes) {
+    return absl::nullopt;
+  }
+
+  return std::make_pair(BytesToBool(*bool_row), std::move(*bytes));
+}
+absl::optional<std::vector<std::pair<bool, std::vector<uint8_t>>>>
+ExtractBoolBytesTupleArray(Span data) {
+  // Get the length of the array.
+  auto array_size_offset_row = ExtractRow(data, 0);
+  if (!array_size_offset_row) {
+    return absl::nullopt;
+  }
+  absl::optional<size_t> offset = BytesToSize(*array_size_offset_row);
+  if (!offset || *offset > data.size()) {
+    return absl::nullopt;
+  }
+  auto array_size_row = ExtractRow(data, (*offset) / kRowLength);
+  if (!array_size_row) {
+    return absl::nullopt;
+  }
+  auto array_size = BytesToSize(*array_size_row);
+  if (!array_size || *array_size > PaddedRowCount(data.size())) {
+    return absl::nullopt;
+  }
+
+  // Increment the offset to the next row since size has been read.
+  *offset += kRowLength;
```

Comment:
Will it work if this part is reused 
```
  // Array is stored as size row and tuple of that size.
  auto [tuple_size, tuple_header] = ExtractArrayInfo(string_array);
  if (!tuple_size)
    return absl::nullopt;
  // Row count in array is reasonable upper limit.
  if (*tuple_size > PaddedRowCount(string_array.size()))
    return absl::nullopt;
  if (*tuple_size == 0) {
    return std::vector<std::string>();
  }
```
from  ExtractStringArray ?

Reply:
I don't think so because at least in this case the `tuple_size` returned by ExtractArrayInfo is not the size of the array.  It's actually the offset pointing to the size of the array.  Here's an example, which is used in a test:

```
// (bytes, bool)[]
// offset of the size of the array
"0000000000000000000000000000000000000000000000000000000000000020"
// size of the array
"0000000000000000000000000000000000000000000000000000000000000001"
// offset of the first element
"0000000000000000000000000000000000000000000000000000000000000020"
// value of bool in the first element
"0000000000000000000000000000000000000000000000000000000000000001"
// offset of the size of the bytes in the first element
"0000000000000000000000000000000000000000000000000000000000000040"
// size of the bytes in the first element
"0000000000000000000000000000000000000000000000000000000000000020"
// first element bytes
"000000000000000000000000000000000000000000000006e83695ab1f893c00"
```

In this case `tuple_size` would be `0000000000000000000000000000000000000000000000000000000000000020`.  We need to get the value at that offset to the actual size.

Reply:
Yeah, ExtractStringArray was to parse string array from struct, but here we actually parsing  one element tuple `(bytes,bool[])` - it is a tuple because it is a returned value. https://docs.soliditylang.org/en/latest/abi-spec.html#function-selector-and-argument-encoding
So read first row, offset within data to its value and ExtractArrayInfo then